src/main/scala/mesosphere/marathon/tasks/TaskOpLogic.scala

```diff
@@ -0,0 +1,15 @@
+package mesosphere.marathon.tasks
+
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.state.AppDefinition
+import org.apache.mesos.{ Protos => Mesos }
+
+/** Infers which TaskOps to crate for given app definitions and offers. */
+trait TaskOpLogic {
```

Comment:
Yes I tried to distinguish the two a little – but I agree, this one's the Factory so I renamed the other one to `...Helper`